apps/admin-page/src/app/pages/order-details/order-details.component.ts

```diff
@@ -0,0 +1,189 @@
+  ];
+
+  constructor(public ordersService: OrdersService, private route: ActivatedRoute, private location: Location) {
+    this.title = 'Loading...';
+  }
+
+  ngOnInit(): void {
+    this.id = this.route.snapshot.paramMap.get('id') || undefined;
+    // Fetching the order.
+    if (this.id !== undefined) {
+      this.fetchOrder(this.id);
+    }
+  }
+
+  /**
+   * Redirects to the previous page.
+   */
+  historyBack(): void {
+    this.location.back();
+  }
+
+  /**
+   * Gets triggered when the status of an order has been changed.\
+   * It will call the API to update the order with its new status.\
+   * In case an error is encountered, the orders will be reloaded from the database.
+   *
+   * @param order - the order containing the new status.
+   */
+  onStatusChange(order: IOrder): void {
+    this.ordersService.updateOrder(order).subscribe({
+      error: (error: HttpErrorResponse) => {
+        if (this.id !== undefined) {
+          this.fetchOrder(this.id);
+        }
+        console.error(error);
+      },
+    });
+  }
+
+  /**
+   * Fetches an order from the API.\
+   * \
+   * Will toggle `isLoading` to `true` while the order is being fetched
+   * and revert it to `false` when it has been fetched.\
+   * Initializes all of the order related values.\
+   * \
+   * **Uses**: `ordersService` -> to call the API.
+   */
+  fetchOrder(id: string): void {
+    this.isLoading = true;
+    this.ordersService.getOrders().subscribe({
+      error: (error: HttpErrorResponse) => {
+        console.error(error);
+      },
+      next: (orders: IOrder[]) => {
+        this.isLoading = false;
+        this.order = orders.find((order) => order.orderId === id);
+        // Setting the page title.
+        this.title = `Order by: ${this.order?.contactInfo.name}`;
+
+        // Customer Information
+        this.emailControl = new FormControl(this.order?.contactInfo.email);
+        this.phoneNumberControl = new FormControl(this.order?.contactInfo.phoneNumber);
+        // Contact Information
+        this.nameControl = new FormControl(this.order?.contactInfo.name);
+        this.cityControl = new FormControl(this.order?.contactInfo.city);
+        this.postcodeControl = new FormControl(this.order?.contactInfo.postcode);
+        this.addressOneControl = new FormControl(this.order?.contactInfo.streetAddress);
+        this.addressTwoControl = new FormControl(this.order?.contactInfo.streetAddress2);
+        // Billing Information
+        this.billingNameControl = new FormControl(this.order?.billingInfo.name);
+        this.billingCityControl = new FormControl(this.order?.billingInfo.city);
+        this.billingPostcodeControl = new FormControl(this.order?.billingInfo.postcode);
+        this.billingAddressOneControl = new FormControl(this.order?.billingInfo.streetAddress);
+        this.billingAddressTwoControl = new FormControl(this.order?.billingInfo.streetAddress2);
+
+        if (this.order?.transactionItems) {
```

Comment:
An order cant Not have at least one transaction item, so this if statement is not needed.